packages/staking/src/features/drawer/StakePoolConfirmation.tsx

```diff
@@ -45,6 +45,26 @@ const isInputSelectionError = (error: any): error is { failure: InputSelectionFa
   Object.hasOwn(error, 'failure') &&
   Object.values(InputSelectionFailure).includes(error.failure);
 
+const calculateDeposit = ({
+  newPortfolio,
+  rewardAccounts,
+  protocolParameters,
+}: {
+  newPortfolio: DelegationPortfolioStakePool[];
+  rewardAccounts?: Wallet.Cardano.RewardAccountInfo[];
+  protocolParameters?: Wallet.Cardano.ProtocolParameters;
+}) => {
+  if (!protocolParameters?.stakeKeyDeposit || !rewardAccounts) return 0;
+
+  const registeredStakeKeysCount = rewardAccounts.filter((account) =>
+    [Wallet.Cardano.StakeKeyStatus.Registered, Wallet.Cardano.StakeKeyStatus.Registering].includes(account.keyStatus)
+  ).length;
+
+  return newPortfolio.length > registeredStakeKeysCount
+    ? (newPortfolio.length - registeredStakeKeysCount) * protocolParameters.stakeKeyDeposit
+    : 0;
```

Comment:
A nitpick: How about always returning the real value here, also the negative one? This way we will have a entire truth on the data level, and later anyway on a UI rendering level there is a check to show only values > 0 so the negative ones won't be shown.
```suggestion
  return (newPortfolio.length - registeredStakeKeysCount) * protocolParameters.stakeKeyDeposit;
```